tools/src/wyvern/target/corewyvernIL/expression/MethodCall.java

```diff
@@ -65,9 +67,17 @@ public String getMethodName() {
 	public List<? extends IExpr> getArgs() {
 		return args;
 	}
+	
+	private ValueType getReceiverType(TypeContext ctx) {
+		if (receiverType == null) {
+			receiverType = objectExpr.typeCheck(ctx);
+		}
+		return receiverType;
+	}
 
 	@Override
 	public ValueType typeCheck(TypeContext ctx) {
+		if (Util.isDynamicType(getReceiverType(ctx))) return Util.dynType();
```

Comment:
Still need to check that the arguments have some type (even if we don't care what it is)